cmake/checks/check_01_cxx_features.cmake

```diff
@@ -99,10 +99,17 @@ ENDIF()
 
 MACRO(_check_cxx_flag _suffix)
   IF("${DEAL_II_CXX_VERSION_FLAG}" STREQUAL "")
-    CHECK_CXX_COMPILER_FLAG("-std=c++${_suffix}" DEAL_II_HAVE_FLAG_stdcxx${_suffix})
-    IF(DEAL_II_HAVE_FLAG_stdcxx${_suffix})
-      SET(DEAL_II_CXX_VERSION_FLAG "-std=c++${_suffix}")
-    ENDIF()
+    IF(CMAKE_CXX_COMPILER_ID MATCHES "MSVC")
+      CHECK_CXX_COMPILER_FLAG("/std:c++${_suffix}" DEAL_II_HAVE_FLAG_stdcxx${_suffix})
+      IF(DEAL_II_HAVE_FLAG_stdcxx${_suffix})
+        SET(DEAL_II_CXX_VERSION_FLAG "/std:c++${_suffix}")
+      ENDIF()
+	ELSE()
```

Comment:
I think you're using 4-space tabs here because the indentation looks funny. Just use spaces instead.